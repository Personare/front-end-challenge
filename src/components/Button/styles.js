import styled from "styled-components";

const ButtonContainer = styled.button`
  background: #2F4A94;
  border-radius: 8px;
  box-shadow: 0 10px 10px 0 rgba(0, 0, 0, 0.2);
  font-size: 1.5rem;
  color: #FFF;

  div {
    width: 100%;
    height: 100%;
    padding: 10px 25px;
    cursor: pointer;
  }
`;

export default ButtonContainer;
