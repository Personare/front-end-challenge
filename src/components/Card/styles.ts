import styled, { css } from "styled-components";

interface ImageProps {
  isMinimal?: boolean;
}
export const Image = styled.img<ImageProps>`
  ${({ isMinimal }) => css`
    height: ${isMinimal ? "150px" : "300px"};
    width: ${isMinimal ? "79px" : "auto"};
    margin-bottom: ${isMinimal ? "10px" : "auto"};
  `}
`;

export const Container = styled.div``;
