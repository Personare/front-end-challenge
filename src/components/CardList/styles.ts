import styled from "styled-components";

export const List = styled.div`
  align-items: center;
  display: grid;
  gap: 20px;
  width: 100%;

  flex-direction: column;
  grid-template-columns: repeat(auto-fill, minmax(162px, 1fr));

  img {
    display: block;
    height: auto;
    width: 100%;
  }
`;
