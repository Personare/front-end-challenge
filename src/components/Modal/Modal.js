import React, { Component } from 'react'
import styled from 'styled-components'
import {
  func, node, arrayOf, oneOfType, string
} from 'prop-types'

class Modal extends Component {
  constructor(props) {
    super(props)
    this.handleKeyUp = this.handleKeyUp.bind(this)
  }

  componentDidMount() {
    window.addEventListener('keyup', this.handleKeyUp, false)
  }

  componentWillUnmount() {
    window.removeEventListener('keyup', this.handleKeyUp, false)
  }

  handleKeyUp(e) {
    const { onClose } = this.props
    if (e.keyCode === 27) {
      e.preventDefault()
      onClose()
      window.removeEventListener('keyup', this.handleKeyUp, false)
    }
  }

  render() {
    const { onClose, children, title } = this.props

    return (
      <Wrapper>
        <ModalOverlay onClick={onClose} />
        <ModalContainer>
          <ModalContent>
            <ModalHeader>
              {title}
              <CloseIcon onClick={onClose} />
            </ModalHeader>
            <ChildrenWrapper>{children}</ChildrenWrapper>
          </ModalContent>
        </ModalContainer>
      </Wrapper>
    )
  }
}

const Wrapper = styled.div`
  visibility: ${props => props.visibilityStatus};
`

const ModalOverlay = styled.div`
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  height: 100%;
  width: 100%;
  z-index: 1000;
  background: rgba(0, 0, 0, 0.76);
`

const ModalContainer = styled.div.attrs({
  'data-modal': 'modal-content'
})`
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 1000;
  overflow: auto;
  text-align: center;
  padding: 4px;
  cursor: pointer;
`

const ModalContent = styled.div.attrs({
  'data-modal': 'modal-dialog'
})`
  position: relative;
  outline: 0;
  width: auto;
  display: inline-block;
  vertical-align: middle;
  box-sizing: border-box;
  cursor: default;
  border-radius: 3px;
  width: auto;
  margin: 30px auto;
  background-color: #ffffff;

  @media (max-width: 575px) {
    min-width: 100%;
    height: 100%;
    overflow: auto;
    margin: 0;
  }
`

const ChildrenWrapper = styled.div`
  display: flex;
  justify-content: center;
`

const ModalHeader = styled.div`
  margin: 10px 0;
  font-size: 18px;
  font-weight: 600;
`

const CloseIcon = styled.img.attrs({
  src: './assets/close_icon.svg'
})`
  width: 30px;
  height: 30px;
  position: absolute;
  right: 0;
  top: 0;
  margin: 5px;
`

Modal.propTypes = {
  onClose: func.isRequired,
  children: oneOfType([arrayOf(node), node]).isRequired,
  title: string
}

Modal.defaultProps = {
  title: ''
}
export default Modal
