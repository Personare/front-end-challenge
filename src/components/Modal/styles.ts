import styled from "styled-components";

export const Overlay = styled.div`
  align-items: center;
  background-color: rgba(0, 0, 0, 0.75);
  display: flex;
  height: 100%;
  justify-content: center;
  left: 0;
  padding: 15px;
  position: fixed;
  top: 0;
  width: 100%;
  z-index: 30;
`;

export const Content = styled.div`
  background-color: white;
  box-shadow: 0 2px 8px rgba(0, 0, 0, 0.25);
  padding: 40px;
  position: relative;
  display: flex;
  flex-direction: column;
  align-items: center;
  gap: 20px;
  width: 350px;
`;

export const Name = styled.h3`
  font-size: 20px;
  line-height: 1;
`;

export const Text = styled.p`
  font-size: 14px;
  line-height: 1.25;
  text-align: justify;
`;

export const CloseButton = styled.button`
  background-color: transparent;
  border: none;
  cursor: pointer;
  position: absolute;
  right: 10px;
  top: 5px;
  line-height: 1;
  font-size: 24px;
  padding: 10px 10px;
  transition: color 200ms ease;

  &:hover {
    color: rgba(0, 0, 0, 0.5);
  }
`;
