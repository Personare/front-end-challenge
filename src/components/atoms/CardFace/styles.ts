import styled from 'styled-components';

type CardFaceProps = {
  rotate: string;
};

export const CardFace = styled.div<CardFaceProps>`
  backface-visibility: hidden;
  height: inherit;
  overflow: hidden;
  position: absolute;
  transform: perspective(200px) rotateY(${({ rotate }) => rotate});
  transition: transform 0.6s linear;
`;

export const CardImage = styled.img`
  height: 100%;
  object-fit: cover;
  object-position: center;
  width: 100%;
`;
