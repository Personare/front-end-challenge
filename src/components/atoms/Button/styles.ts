import styled from 'styled-components';
import { colors } from 'styles/variables';

export const Button = styled.button`
  background: ${colors.pink};
  border: 0;
  border-bottom: 4px solid rgba(0, 0, 0, 0.3);
  border-radius: 8px;
  box-shadow: 0 4px 8px ${colors.gray};
  color: ${colors.white};
  cursor: pointer;
  font-family: 'Open Sans', sans-serif;
  font-size: 18px;
  font-weight: 600;
  letter-spacing: 0.05em;
  padding: 12px 24px;

  :hover {
    opacity: 0.9;
  }

  &[disabled] {
    cursor: initial;
    opacity: 0.5;
  }
`;
