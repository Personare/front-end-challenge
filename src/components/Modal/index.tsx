import { ICard } from "../../config/interfaces";
import Card from "../Card";
import { Content, Text, Overlay, Name, CloseButton } from "./styles";

interface IModal {
  card: ICard;
  onClose: () => void;
}

const Modal: React.FC<IModal> = ({ card, onClose }) => {
  const handleContentClick = (e: any) => {
    e.stopPropagation();
  };

  return (
    <Overlay onClick={onClose} data-cy="modal-overlay">
      <Content onClick={handleContentClick}>
        <Name>{card.name}</Name>
        <Card card={card} />
        <Text>
          The standard chunk of Lorem Ipsum used since the 1500s is reproduced
          below for those interested. Sections 1.10.32 and 1.10.33 from "de
          Finibus Bonorum et Malorum" by Cicero.
        </Text>

        <CloseButton onClick={onClose}>&times;</CloseButton>
      </Content>
    </Overlay>
  );
};
export default Modal;
