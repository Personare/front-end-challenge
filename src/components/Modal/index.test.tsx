import { shallow } from "enzyme";
import Modal from ".";
import { ICard } from "../../config/interfaces";
import { CloseButton, Overlay } from "./styles";

const CARD: ICard = {
  name: "Name Test Modal",
  image: "Image Test Modal",
  show: false,
};

describe("Modal", () => {
  it("should match snapshot", () => {
    const onCloseMock = jest.fn();
    const wrapper = shallow(<Modal card={CARD} onClose={onCloseMock} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should call the function when closing the modal", () => {
    const onCloseMock = jest.fn();
    const wrapper = shallow(<Modal card={CARD} onClose={onCloseMock} />);

    wrapper.find(CloseButton).simulate("click");
    expect(onCloseMock).toHaveBeenCalledTimes(1);

    wrapper.find(Overlay).simulate("click");
    expect(onCloseMock).toHaveBeenCalledTimes(2);
  });
});
