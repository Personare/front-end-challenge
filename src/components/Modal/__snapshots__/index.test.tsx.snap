// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should match snapshot 1`] = `
<styled.div
  onClick={[MockFunction]}
>
  <styled.div
    onClick={[Function]}
  >
    <styled.h3>
      Name Test Modal
    </styled.h3>
    <Card
      card={
        Object {
          "image": "Image Test Modal",
          "name": "Name Test Modal",
          "show": false,
        }
      }
    />
    <styled.p>
      The standard chunk of Lorem Ipsum used since the 1500s is reproduced below for those interested. Sections 1.10.32 and 1.10.33 from "de Finibus Bonorum et Malorum" by Cicero.
    </styled.p>
    <styled.button
      onClick={[MockFunction]}
    >
      Ã—
    </styled.button>
  </styled.div>
</styled.div>
`;
